### PR TITLE
Detect whether the k8s distribution is hypercube

### PR DESCRIPTION
**Pull request by ruettimac**
Detect whether the k8s distribution is hypercube or not and adapts the process names accordingly.

### CODE REVIEW

File: libraries/kubernetes.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  name 'kubernetes'
+  desc 'Custom resource which abstracts the various kubernetes runtimes like hyperkube'
+
+  def initialize
+    @file = inspec.file('/usr/bin/hyperkube')
+    @ishyperkube = @file.file?
+    Log.debug("The installation is hyperkube=#{@ishyperkube}")
+  end
+
+  def hyperkube
```
Comment (chris-rock) [inline libraries/kubernetes.rb]: we should name this `hyperkube?`

File: controls/1_3_master_node_controller_manager.rb [hunk 2]
```diff
@@ -17,8 +17,10 @@
 
 title '1.3 Master Node: Controller Manager'
 
+controller_manager = attribute('controller_manager', default: kubernetes.hyperkube ? 'controller-manager' : 'kube-controller-manager', description: 'The name of the controller manager process')
```
Comment (chris-rock) [inline controls/1_3_master_node_controller_manager.rb]: Wouldn't it be better to have a method called `kubernetes.controller_manager` where the method returns the right value?

File: controls/1_1_master_node_api_server.rb [hunk 3]
```diff
@@ -17,8 +17,10 @@
 
 title '1.1 Master Node: API Server'
 
+apiserver = attribute('apiserver', default: kubernetes.hyperkube ? 'apiserver' : 'kube-apiserver', description: 'The name of the apiserver process')
```
Comment (chris-rock) [inline controls/1_1_master_node_api_server.rb]: I think we should return the command from `kubernetes.apiserver` and abstract the decision making there. Similar to what we do for ssh or nginx https://github.com/dev-sec/nginx-baseline/blob/master/libraries/nginx_lib.rb#L24-L37

File: controls/1_3_master_node_controller_manager.rb (hunk 2 above)
Comment (ruettimac) [reply controls/1_3_master_node_controller_manager.rb]: Good point! So I will then abstract the logic of the process name to return for the various components in the kubernetes class. The same applies to the next comment.

File: controls/1_1_master_node_api_server.rb (hunk 3 above)
Comment (ruettimac) [reply controls/1_1_master_node_api_server.rb]: See comment above

File: libraries/kubernetes.rb (hunk 1 above)
Comment (ruettimac) [reply libraries/kubernetes.rb]: I think this will then be obsolete when the abstract the decision in kubernetes class - right?

Comment (chris-rock) [reply libraries/kubernetes.rb]: Right, I think we can get away without this for now. We only need this if you think there are cases where test output will be different for hyperkube. But we could add this later if really required.

**Comment by ruettimac** [comment]
Refactoring done. Since I am not a Ruby guru - kubernetes.rb smells for refactoring. Maybe you have some advices?

**Comment by rarenerd** [comment]
@chris-rock can you have another look at this now your comments have been addressed?

**Pull request merged**